src/test/java/htsjdk/samtools/ValidateSamFileTest.java

```diff
@@ -150,29 +150,32 @@ public void testUnpairedRecords() throws IOException {
         Assert.assertEquals(results.get(SAMValidationError.Type.INVALID_UNPAIRED_MATE_REFERENCE.getHistogramString()).getValue(), 1.0);
     }
 
-    @Test
-    public void testPairedRecords() throws IOException {
+    @Test(dataProvider = "Topologies")
+    public void testPairedRecords(final SAMSequenceRecord.Topology topology) throws IOException {
         final SAMRecordSetBuilder samBuilder = new SAMRecordSetBuilder();
 
-        for (int i = 0; i < 5; i++) {
+        final SAMSequenceDictionary dict = samBuilder.getHeader().getSequenceDictionary();
+        for (int i = 0; i < 6; i++) {
             samBuilder.addPair(String.valueOf(i), i, i, i + 100);
+            dict.getSequence(i).setTopology(topology);
         }
         final Iterator<SAMRecord> records = samBuilder.iterator();
         records.next().setMateReferenceName("*");
-        records.next().setMateAlignmentStart(Integer.MAX_VALUE);
+        records.next().setMateAlignmentStart(Integer.MAX_VALUE); // mate start off the end
```

Comment:
My understanding of the spec is that POS≤LEN must still be upheld, so this doesn't seem right.